src/features/applications/components/application-details.tsx

```diff
@@ -109,45 +122,56 @@ export function ApplicationDetails({ application }: Props) {
     ]
   ).filter(isDefined)
 
+  const abiMethodDefinitions = useLoadableApplicationAbiMethodDefinitions(application)
+
   return (
-    <div className={cn('space-y-4')}>
+    <div className="space-y-4">
       <Card aria-label={applicationDetailsLabel}>
         <CardContent>
-          <div className={cn('flex gap-2')}>
+          <div className="flex gap-2">
             <DescriptionList items={applicationItems} />
-            <OpenJsonViewDialogButton json={application.json} expandJsonLevel={expandApplicationJsonLevel} />
+            <div className="ml-auto hidden flex-col-reverse justify-end gap-2 md:flex lg:flex-row">
+              <ApplicationProgramsButton application={application} />
+              <OpenJsonViewDialogButton json={application.json} expandJsonLevel={expandApplicationJsonLevel} />
+            </div>
           </div>
         </CardContent>
       </Card>
-      <Card aria-label={applicationApprovalProgramLabel}>
-        <CardContent className={cn('space-y-1')}>
-          <h2>{applicationApprovalProgramLabel}</h2>
-          <ApplicationProgram tabsListAriaLabel={applicationApprovalProgramTabsListAriaLabel} base64Program={application.approvalProgram} />
-        </CardContent>
-      </Card>
-      <Card aria-label={applicationClearStateProgramLabel}>
-        <CardContent className={cn('space-y-1')}>
-          <h2>{applicationClearStateProgramLabel}</h2>
-          <ApplicationProgram
-            tabsListAriaLabel={applicationClearStateProgramTabsListAriaLabel}
-            base64Program={application.clearStateProgram}
-          />
-        </CardContent>
-      </Card>
-      <Card aria-label={applicationGlobalStateLabel}>
-        <CardContent className={cn('space-y-1')}>
-          <h2>{applicationGlobalStateLabel}</h2>
-          <ApplicationGlobalStateTable application={application} />
-        </CardContent>
-      </Card>
-      <Card aria-label={applicationBoxesLabel}>
-        <CardContent className={cn('space-y-1')}>
-          <h2>{applicationBoxesLabel}</h2>
-          <ApplicationBoxes applicationId={application.id} />
+      <RenderLoadable loadable={abiMethodDefinitions}>
```

Comment:
This loader looks a little strange to me. I think we'd be better off simply not showing any loading state at this point.
 
Potentially a better way in the future is to render the card and have some messaging pointing back to app registry/lab on how to add your app spec. It only makes sense to do this once we have activated the app registry/lab feature.